It is assumed that the mean systolic blood pressure is μ = 120 mm Hg. In the Honolulu Heart Study, a sample of n = 100 people 
had an average systolic blood pressure of 130.1 mm Hg with a standard deviation of 21.21 mm Hg. 
Is the group significantly different (with respect to systolic blood pressure!) from the regular population?

1. State Null Hypothesis and Alternative Hypothesis
2. Gather known data / variables e.g. population (n), confidence and significance levels ( 1-alfa and alfa)
3. Choose statistical test : 
    a) compute test statistic and compare to critical value with a confidence level
    b) compare p-value against significance level
        Criterias : IF n < 30 --> choose t distribution ; ELSE --> choose normal distribution
4. 

1.
<h3> Hypothesis H0:</h3> <h2>People from Honolulu have a much higher blood pressure than average population</h2> 
<h3> Hypothesis Ha:</h3> <h2>People from Honolulu has an average higher blood pressure </h2>

2. mean_pop = 120 mm Hg ; n = 100 ; mean_h = 130 mm Hg ; stnd.dev = 21.21 mm Hg ; alfa = 0.05% ; confidence level = 0.95 %

Z = (mean_pop - mean_h) / ( stnd.dev / 50^(1/2) ) = 120 - 130 / (21.21 / (100^(1/2) ) = -10 / 2.121  = -4.71

In [1]:
import scipy.stats as stats
import pandas as pd
import math

In [2]:
zc = +-stats.norm.ppf(1-(0.05))

In [3]:
zc

-1.6448536269514722

-1.64 > -4.71 --> Confirm null Hypothesis

In a packing plant, a machine packs cartons with jars. It is supposed that a new machine will pack faster on the average
than the machine currently used. To test that hypothesis, the times it takes each machine to pack ten cartons are recorded. 
The results, in seconds, are shown in the tables in the file Data/machine.txt. Assume that there is sufficient evidence to
 conduct the t test, does the data provide sufficient evidence to show if one machine is better than the other

In [4]:

df = pd.read_csv("Data/Data_Machine.txt")


In [5]:
df

New Machine\t  Old Machine
0         42.1\t        42.7
1         41\t          43.6
2         41.3\t        43.8
3         41.8\t        43.3
4         42.4\t        42.5
5         42.8\t        43.5
6         43.2\t        43.1
7         42.3\t        41.7
8           41.8\t        44
9         42.7\t        44.1

In [6]:
df['New Machine\t  Old Machine'].value_counts()

42.1\t        42.7    1
41\t          43.6    1
41.3\t        43.8    1
41.8\t        43.3    1
42.4\t        42.5    1
42.8\t        43.5    1
43.2\t        43.1    1
42.3\t        41.7    1
41.8\t        44      1
42.7\t        44.1    1
Name: New Machine\t  Old Machine, dtype: int64

In [7]:
df['New'], df['Old'] = df['New Machine\t  Old Machine'].str.split('\t', 1).str
    

C:\Users\Seinf\AppData\Local\Temp/ipykernel_11572/312059950.py:1: FutureWarning: Columnar iteration over characters will be deprecated in future releases.
  df['New'], df['Old'] = df['New Machine\t  Old Machine'].str.split('\t', 1).str


In [8]:
df = df.iloc[: , 1:]

In [9]:
df

New             Old
0  42.1            42.7
1    41            43.6
2  41.3            43.8
3  41.8            43.3
4  42.4            42.5
5  42.8            43.5
6  43.2            43.1
7  42.3            41.7
8  41.8              44
9  42.7            44.1

In [10]:
df['New']  = df['New'].astype(float)

In [11]:
df['Old']  = df['Old'].astype(float)

In [12]:
mean1 = df['New'].mean()
mean1

42.14

In [13]:
mean2 = df.Old.mean()
mean2

43.230000000000004

In [14]:
s1 = df['New'].std()

In [15]:
s2 = df['Old'].std()

In [16]:
t = (mean1 - mean2) / math.sqrt ( ((s1**2) / 2 ) + ((s2**2) / 2) ) 

In [17]:
t

-1.5192877588257143

In [18]:
tc = stats.t.ppf(1-(0.05/2),9)

In [19]:
tc

2.2621571627409915

In [20]:
abs(t)

1.5192877588257143

t< TC -> accept the null hypothesis, new machine will pack faster